packages/contracts/contracts/LenderCommitmentForwarder.sol

```diff
@@ -334,6 +337,77 @@ contract LenderCommitmentForwarder is TellerV2MarketForwarder {
         uint16 _interestRate,
         uint32 _loanDuration
     ) external returns (uint256 bidId) {
+        require(
+            commitments[_commitmentId].collateralTokenType <=
+                CommitmentCollateralType.ERC1155_ANY_ID,
+            "Invalid commitment collateral type"
+        );
+
+        _acceptCommitment(
+            _commitmentId,
+            _principalAmount,
+            _collateralAmount,
+            _collateralTokenId,
+            _collateralTokenAddress,
+            _interestRate,
+            _loanDuration
+        );
+    }
+
+    function acceptCommitmentWithProof(
+        uint256 _commitmentId,
+        uint256 _principalAmount,
+        uint256 _collateralAmount,
+        uint256 _collateralTokenId,
+        address _collateralTokenAddress,
+        uint16 _interestRate,
+        uint32 _loanDuration,
+        bytes32[] calldata _merkleProof
+    ) external returns (uint256 bidId) {
+        bytes32 _merkleRoot = bytes32(
+            commitments[_commitmentId].collateralTokenId
+        );
+        bytes32 _leaf = keccak256(abi.encodePacked(_collateralTokenId));
+
+        //make sure collateral token id is a leaf within the proof
+        require(
+            MerkleProofUpgradeable.verifyCalldata(
+                _merkleProof,
+                _merkleRoot,
+                _leaf
+            ),
+            "Invalid proof"
+        );
+
+        require(
+            commitments[_commitmentId].collateralTokenType ==
+                CommitmentCollateralType.ERC721_MERKLE_PROOF ||
+                commitments[_commitmentId].collateralTokenType ==
+                CommitmentCollateralType.ERC1155_MERKLE_PROOF,
+            "Invalid commitment collateral type"
+        );
```

Comment:
I think it makes more sense to check the `collateralTokenType` before we verify the merkle proof